### PR TITLE
Update documentation to reflect actual implementation (fix #281)

### LINKED ISSUE #281: Not inspecting surplus arguments
While the API doc for the logging methods describes
> ... If more args follows msg, these will be used to format msg using util.format.  

The actual behaviour is different when sending more arguments than interpolation placeholders. While console.log, util.format and the npm module 'debug' all behave similarly (concatenating with space these extras after running them through util.inspect), pino simply truncates these extra arguments - tools.js:94:
`        n.length = l + countInterp(a, '%d') + countInterp(a, '%s') + 1
`  
Doesn't necessarily feel like a speed optimization since it just blocks this use case (to avoid these objects being inspected, don't send them in the first place - same result).  
If this makes sense to you too, would the correct logic to apply in genLog be: if no %j and countInterp(%j,%d,%s) <= n.length we can call quick escape, otherwise pass all of n into util.format.

What do you think?

### PR DESCRIPTION
**Pull request by jsumners**
As detailed in https://github.com/pinojs/pino/issues/281#issuecomment-379248468 this PR fixes issue #281.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -329,8 +325,7 @@ If more args follows `msg`, these will be used to format `msg` using
 ### Discussion:
 Log at `'warn'` level the given `msg`. If the first argument is an
 object, all its properties will be included in the JSON line.
-If more args follows `msg`, these will be used to format `msg` using
-[`util.format`][util-format].
```
[the remark this answers is not in the document]
Comment (jsumners) [reply docs/API.md]: Let's look at the documentation in whole for one method:

```js
<a id="fatal"></a>
## .fatal([obj], msg, [...])

### Parameters:
+ `obj` (object): object to be serialized
+ `msg` (string): the log message to write
+ `...` (*): format string values when `msg` is a format string

### Discussion:
Log at `'fatal'` level the given `msg`. If the first argument is an
object, all its properties will be included in the JSON line.
If more args follow `msg`, they will be silently ignored.
```

We aren't even documenting the `.fatal('%s%s', 'a', 'b')` form of the function invocation. Thus, in relation to the rest of the documentation, this PR is correct.

The issue is that these methods accept an option *first position* parameter. That makes documenting the various use cases difficult. Do we have any suggestions for alternate documentation? Or should we stick with what we have?

[the remark this answers is not in the document]
Comment (mcollina) [reply docs/API.md]: @davidmarkclements is the man for format strings. If it was for me we wouldn't have added them at all.

[the remark this answers is not in the document]
Comment (jsumners) [reply docs/API.md]: Well I'm happy he got them added then. I use them all the time :)

[the remark this answers is not in the document]
Comment (mcollina) [reply docs/API.md]: So we should document those, not remove their doc.

[the remark this answers is not in the document]
Comment (jsumners) [reply docs/API.md]: Updated.

**Pull request merged**